test/auth/auth.test.js

```diff
@@ -0,0 +1,63 @@
+const chai = require('chai'),
+  { expect } = chai;
+const chaiHttp = require('chai-http');
+const userModelManager = require('../../lib/modelManagers/usermodel');
+const authController = require('../../controllers/AuthController');
+const server = require('../../index');
+
+chai.use(chaiHttp);
+chai.should();
+
+let newUser;
+
+const next = () => {};
+const res = () => {};
+
+/*
+ Test to make sure User is created and token is generated
+*/
+
+describe('Test for adding a new user and add token to request', () => {
+  describe('Test with correct information', () => {
+    const req = {
+      body: {
+        email: 'testmail@gmail.com',
+        password: '111111',
+        firstname: 'testfirstname',
+        lastname: 'testlastname'
+      }
+    };
+    const correctDetails = {
+      email: 'testmail@gmail.com',
+      password: '111111'
+    };
+
+    it('Should add a new user to the database', async () => {
+      await authController.signUp(req, res, next);
+    });
+    it('Should check if added user exist', async () => {
+      newUser = await userModelManager.findUser(req.body.email);
```

Comment:
`newUser` local variable is more prefered over a global variable as it makes code easy to follow and maintainable.

Reply:
i have changed that